### PR TITLE
Adds an example of testing spring webflux application with kotest

### PR DESCRIPTION
**Pull request by sharmabhawna**
This example uses kotest framework and [springMockk](https://github.com/Ninja-Squad/springmockk) for testing a rest controller written using spring webflux.

### CODE REVIEW

File: kotest-examples/kotest-examples-spring-webflux/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+//   filter {
+//      setFailOnNoMatchingTests(false)
+//   }
+//   testLogging {
+//      showExceptions = true
+//      showStandardStreams = true
+//      events = setOf(org.gradle.api.tasks.testing.logging.TestLogEvent.FAILED, org.gradle.api.tasks.testing.logging.TestLogEvent.PASSED)
+//      exceptionFormat = org.gradle.api.tasks.testing.logging.TestExceptionFormat.FULL
+//   }
+//}
```
Comment (LeoColman) [inline kotest-examples/kotest-examples-spring-webflux/build.gradle.kts]: Why this chunk of commented code?

**Review by LeoColman — CHANGES_REQUESTED** [review]

**Comment by LeoColman** [comment]
Does this work out of the box, without the spring listener?

**Comment by sharmabhawna** [comment]
> Does this work out of the box, without the spring listener?

@Kerooker We are using ```@AutoConfigureWebTestClient``` and ```@SpringbootTest``` which manages constructor injection. We don't need to register kotest spring listener we only need it to be present in classpath so that we dont get the following error.

```bash
java.lang.IllegalArgumentException: wrong number of arguments
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
	at io.kotest.core.engine.InstantiateSpecKt.javaReflectNewInstance(instantiateSpec.kt:21)
	at io.kotest.core.engine.InstantiateSpecKt.instantiateSpec(instantiateSpec.kt:15)
	at io.kotest.core.engine.SpecExecutor.createInstance(SpecExecutor.kt:94)
	at io.kotest.core.engine.SpecExecutor.execute(SpecExecutor.kt:35)
	at io.kotest.core.engine.KotestEngine$submitBatch$$inlined$forEach$lambda$1$1.invokeSuspend(KotestEngine.kt:66)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:274)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:79)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:54)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:36)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at io.kotest.core.engine.KotestEngine$submitBatch$$inlined$forEach$lambda$1.run(KotestEngine.kt:65)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:830)
```

You can run the following command to check that test are running fine.
```bash
./gradlew :kotest-examples:kotest-examples-spring-webflux:test
```

File: kotest-examples/kotest-examples-spring-webflux/build.gradle.kts (hunk 1 above)
Comment (sharmabhawna) [reply kotest-examples/kotest-examples-spring-webflux/build.gradle.kts]: I will remove it.

**Comment by ashishkujoy** [comment]
@Kerooker Tests are running fine, is it OK merge?

**Comment by LeoColman** [comment]
Sure

**Pull request merged**